docs/design_docs/cached_outputs.rst

```diff
@@ -0,0 +1,343 @@
+
+Date: 2021/07/30
+
+Contributors: Xylar Asay-Davis
+
+Summary
+-------
+
+We would like to have a way to download output files for ``compass`` steps from
+an online cache instead of generating them each time the step runs.  The
+primary motivation for this is to optionally avoid time-consuming steps for
+generating meshes and initial conditions for faster regression testing with
+MPAS components in "forward" mode.  Potential other uses could include cached
+results as baselines for validation.  A challenge for this capability is
+providing an easy way for both developers and users to control which steps in a
+test case or suite are cached and which are run as normal.
+
+
+Requirements
+------------
+
+.. _req_cached:
+
+Requirement: cached outputs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/07/30
+
+Contributors: Xylar Asay-Davis
+
+Each ``compass`` step defines its output files in the ``compass.Step.outputs``
+attribute. For selected steps (see :ref:`req_select`), we require a mechanism
+to download cached files for each of these outputs and to use these cached
+files for the outputs of the step instead of computing them.  
+
+.. _req_select:
+
+Requirement: selecting whether to use cached outputs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/07/30
+
+Contributors: Xylar Asay-Davis
+
+There needs to be a mechanism for developers and users to select which steps
+are run as normal and which use cached outputs.  For this mechanism to be
+practical, it should not be overly tedious or manual (e.g. manually setting a 
+flag for each step).
+
+.. _req_update:
+
+Requirement: updating cached outputs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/07/30
+
+Contributors: Xylar Asay-Davis
+
+There should be a documented process for creating cached outputs for steps and
+uploading them.
+
+.. _req_unique:
+
+Requirement: unique identifier for cached outputs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/07/30
+
+Contributors: Xylar Asay-Davis
+
+There should be a mechanism for giving each cached output file a unique 
+identifier (such as a date stamp).  A given version (git hash or release) of 
+``compass`` should know which cached files to download.  Older cached files
+should be retained so that older versions of ``compass`` can still be used
+with these cached files.  
+
+.. note::
+
+    It may be worthwhile to include a process for deprecating and then deleting
+    old cache files.
+
+.. _req_normal_or_cached:
+
+Requirement: either "normal" or "cached" versions of a step
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/07/30
+
+Contributors: Xylar Asay-Davis
+
+We **do not** require the ability to set up a "normal" and a "cached" version
+of the same step within a ``compass`` test case or suite.  (If this is not the
+case, it would place important constraints on the design solution.)
```

Comment:
This is the main requirement where I would like feedback.  Do we foresee needing both "normal" and "cached" version of steps?  If so, how do we avoid having a second copy of every test case where we want to have cached outputs (and every test case that depends on the "cached" test case)?  If we don't need that ability, the design I have here should serve us well.